src/main/java/com/visenze/datatracking/Tracker.java

```diff
@@ -0,0 +1,178 @@
+package com.visenze.datatracking;
+
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+import com.google.gson.Gson;
+import com.visenze.datatracking.data.DataCollection;
+import com.visenze.datatracking.data.DataTrackingResponse;
+import com.visenze.datatracking.data.DeviceData;
+import com.visenze.datatracking.data.Event;
+import com.visenze.datatracking.data.EventsBody;
+import com.visenze.datatracking.data.ResultData;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class Tracker {
+
+    public static final String TAG = "Tracker";
+    private TrackingService trackingService;
+    private String code;
+    private String uid;
+    private String sid;
+    private DeviceData mDeviceData;
+    private DataCollection dataCollection;
+
+    public Tracker(String code, String uid, String sid, boolean isCN, DataCollection dataCollection) {
+        this.code = code;
+        this.uid = uid;
+        this.sid = sid;
+        if (isCN) {
+            trackingService = HttpInstance.getRetrofitInstance(Constants.BASE_URL_CN).create(TrackingService.class);
+        } else {
+            trackingService = HttpInstance.getRetrofitInstance(Constants.BASE_URL).create(TrackingService.class);
+        }
+
+        this.dataCollection = dataCollection;
+    }
+
+    private boolean isValidResponse(Response<DataTrackingResponse> response) {
+        return (response.isSuccessful() && response.body() != null && "OK".equals(response.body().getStatus()));
```

Comment:
this is not valid for the image tracker. No need to check for "OK".equals() . If there is an error checking http status is enough.

Reply:
single event send api does not use this check. only batch event use this check.